README.md

```diff
@@ -247,6 +248,107 @@ of how some key tasks can be carried out using Valet:
 - [NPM](https://docs.npmjs.com/cli) commands may be executed by using `npm <command>`.
 - [Artisan](https://laravel.com/docs/9.x/artisan) commands may be executed by using `php artisan <command>`.
 
+
+### Local development setup using docker compose:
+1. Install docker according to your platform instructions found [here](https://docs.docker.com/get-docker/).
+2. Clone the repository:
+
+    ```bash
+    git clone https://github.com/accessibility-exchange/platform.git
+    cd platform
+    ```
+
+3. Create a `.env` file from the included example file:
+
+    ```bash
+    cp .env.local.example .env
+    ```
+   
+    Then, change the `APP_ENV` value to `local`:
+
+    ```dotenv
+    APP_ENV=local
+    ```
+
+4. Build you application container that will be used to generate keys:  
+
+    ```bash
+    docker compose -f docker-compose.local.yml build platform.test
+    ```
+
+5. Generate an encryption key for [CipherSweet](https://github.com/spatie/laravel-ciphersweet):
+
+    ```bash
+    docker run --rm --entrypoint '' platform-platform.test openssl rand -hex 32
+    ```
+
+    Add it to your `.env` file:
+    
+    ```dotenv
+    CIPHERSWEET_KEY="<your key>"
+    ```
+
+6.  Generate an application key:
+
+    ```bash
+    docker run --rm --entrypoint '' platform-platform.test php artisan key:generate --show
+    ```
+
+    Add it to your `.env` file:
+    
+    ```dotenv
+    APP_KEY="<your key>"
+    ```
+
+7. Generate your database password:
+
+    ```bash
+    docker run --rm --entrypoint '' platform-platform.test openssl rand -hex 32
+    ```
+    
+    Add it to your `.env` file:
+    
+    ```dotenv
+    DB_PASSWORD="<your key>"
+    ```
+
+8. Alter the numerical IDs that PHP will run as in the application container:
+    Reason: your local directories will be mapped into the application container to allow your changes to be viewed in real time.
+
+    Find your local user ID & GROUP (Linux & MacOS):
+
+    ```bash
+    ls -ln
+    ```
+
+    You will see output like below. In the below case user is `1000` and group id is `1001`.
+
+    ```bash
+    total 1124
+    drwxr-xr-x 18 1000 1001   4096 Mar 20 12:56 app
+    -rwxr-xr-x  1 1000 1001   1686 Nov  2 12:10 artisan
+    ```
+
+    Add them to your `.env` file:
+
+    ```dotenv
+    WWWUSER=<your user id>
+    WWWGROUP=<your group id>
+    ```
+
+9.  Start up the entire stack:
+   
+   ```bash
+   docker compose -f docker-compose.local.yml up -d
+   ```
```

Comment:
I can't seem to connect to the web application. What is the URL I should be using? It's probably worth mentioning in the docs too.

Reply:
You need to use **https**://localhost in order to visit. There is an SSL proxy container with a self signed certificate because otherwise some CSS files don't load because they are being requested on `https`. 

What I do is create an entry in my `/etc/hosts` file of `127.0.0.1 platform.test`. Then I visit using [https://platform.test](https://platform.test).